### PR TITLE
[#452] Support folder brokerProperties mounts

### LINKED ISSUE #457: extra mount watch is too simplistic and potentially costly
To track the status of referenced resources in #429 a new watch was added on config maps and secrets.

https://github.com/artemiscloud/activemq-artemis-operator/commit/c6f69135df6c6b583c5bcc4d7c683f0b2930b95c#diff-4b39fd9a9f5927e7fa6934d6892919f2ef47e81568ec6908feafcac119d52347

The filter tracks the extramounts on our CRs, even with no extramounts, we still have the watch events.

it would be better if we could have a watch per cr extra mount, 100s of watches rather than 10000s of watch events.

We would need to find a good stateless way to map watchers to CRs.extramounts such that reconcile is idempotent

### PR DESCRIPTION
**Pull request by ruromero**
Fix #457

### CODE REVIEW

**Review by gtully — APPROVED** [review]
looks good, thanks!

File: controllers/activemqartemis_reconciler.go [hunk 1]
```diff
@@ -1989,12 +1979,44 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
...
+		Value: brokerPropertiesConfigFileSystemPropValue(mountPoint, resourceName, brokerPropertiesMapData),
+	}
+	environments.CreateOrAppend(podSpec.InitContainers, &javaOpts)
+
+	// older versions won't use with JAVA_ARGS_APPEND and these will prevale over JAVA_OPTS
+	javaArgsAppend := corev1.EnvVar{
+		Name:  "JAVA_ARGS_APPEND",
+		Value: brokerPropertiesConfigFolderSystemPropValue(mountPoint, resourceName, brokerPropertiesMapData),
+	}
+	environments.CreateOrAppend(podSpec.Containers, &javaArgsAppend)
```
Comment (brusdev) [inline controllers/activemqartemis_reconciler.go:2002]: Why do we need `brokerPropertiesConfigFolderSystemPropValue`? Does `brokerPropertiesConfigFileSystemPropValue` work with older and newer versions?

Comment (gtully) [reply controllers/activemqartemis_reconciler.go:2002]: 2.27.1 introduced support for treating the brokerProperties url ending in a / as a directory. That version also introduced JAVA_ARGS_APPEND.
So the 'folder' via the new env var is what we want, it gives us freedom to ignore the name or the properties file. But for existing deployments that just mount a single immutable properties file, we need to respect the use of the file url.

Comment (gtully) [reply controllers/activemqartemis_reconciler.go:2002]: yes, brokerPropertiesConfigFileSystemPropValue works with old and new.

Comment (brusdev) [reply controllers/activemqartemis_reconciler.go:2002]: I see, I was wondering that a single implementati on could be more easy to maintain if both implementations are equivalent, anyway I have no strong opinioni on this, It LGTM thanks for this fix

**Review by brusdev — APPROVED** [review]

**Comment by brusdev** [comment]
I'll revert this PR because `JAVA_ARGS_APPEND` is not expanded causing the following warning in the broker log:
```
WARN  [org.apache.activemq.artemis.core.server] AMQ222219: File /amq/extra/secrets/aa-controller6472a-props/broker-${STATEFUL_SET_ORDINAL} does not exist
```

Execute `ordinal broker properties` to reproduce this issue.

**Comment by brusdev** [comment]
According to the [EnvVar v1 core reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#envvar-v1-core):
```
Variable references $(VAR_NAME) are expanded using the previous defined environment variables in the container and any service environment variables. If a variable cannot be resolved, the reference in the input string will be unchanged. The $(VAR_NAME) syntax can be escaped with a double $$, ie: $$(VAR_NAME). Escaped references will never be expanded, regardless of whether the variable exists or not. Defaults to "".
```

We have 2 options to fix this issue:

1. Change the artemis script adding `JAVA_ARGS_APPEND=`eval "echo $JAVA_ARGS_APPEND"``
2. Add `{name: PODNAME, valueFrom: {fieldRef: {fieldPath: metadata.name}}}` to the container env and replace `${STATEFUL_SET_ORDINAL}` with `$(PODNAME)`

**Comment by gtully** [comment]
new PR https://github.com/artemiscloud/activemq-artemis-operator/pull/482

**Pull request merged**